site/source/docs/api_reference/Filesystem-API.rst

```diff
@@ -20,6 +20,21 @@ The automatic tests in `tests/test_core.py <https://github.com/emscripten-core/e
 
 A high level overview of the way File Systems work in Emscripten-ported code is provided in the :ref:`file-system-overview`.
 
+New File System: WasmFS
+=======================
+
+.. note:: Current Status: Work in Progress
+
+WasmFS is a high-performance, fully-multithreaded, WebAssembly-based file system layer for Emscripten that will replace the existing JavaScript version.
+
+The JavaScript-based file system was originally written by a community member. At that point thread support did not exist and using JavaScript was better optimized compared to compiling from C. Eventually, with the arrival and usage of pthreads, the file system was modified to proxy all I/O to the main thread.
+
+The new file system should solve the poor performance issues caused by proxying to the main thread and also provide full multi-threading support. Additionally, WasmFS will provide support for users to define their own backend storage options using a standardized, modular backend API. Find out more by reading the design doc or Github tracking issue.
```

Comment:
```suggestion
The JavaScript-based file system was originally written before pthreads were supported and when it was more optimal to write code in JS. As a result it has overhead in pthreads builds because we must proxy to the main thread where all filesystem operations are done. WasmFS, instead, is compiled to wasm and has full multithreading support. It also aims to be more modular and extensible.

Find out more by reading the design doc or Github tracking issue.
```

Reply:
(Trying to shorten this a little, and not get into too many historical details.)

Reply:
Yep sounds good!